File: sensor.js
/**
 * Sensor.
 * @class
 */
class Sensor {
  /**
   * Constructor.
   * @constructor
   *
   * @param {integer} lowerLimit - Lowest valid buffer value.
   * @param {integer} upperLimit - Highest valid buffer value.
   * @param {array<integer>} buffer - Array of sensor data.
   */
  constructor(lowerLimit, upperLimit, buffer) {
  }

  /**
   * The average value from the internal buffer.
   *
   * @returns {integer}
   */
  mean() {
  }

  /**
   * The most frequently occurring value in the internal buffer.
   *
   * @returns {integer}
   */
  mode() {
  }

  /**
   * The middle value from the internal buffer.
   *
   * @returns {integer}
   */
  median() {
  }
}

module.exports = Sensor;

File: sensor.rb
# Sensor.
class Sensor
  # Initialize.
  #
  # @param lower_limit [Integer] Lowest valid buffer value.
  # @param upper_limit [Integer] Highest valid buffer value.
  # @param buffer [Array<Integer>] Array of sensor data.
  def initialize(lower_limit, upper_limit, buffer)
  end

  # The average value from the internal buffer.
  #
  # @return [Integer]
  def mean
  end

  # The most frequently occurring value in the internal buffer.
  #
  # @return [Integer]
  def mode
  end

  # The middle value from the internal buffer.
  #
  # @return [Integer]
  def median
  end
end
